standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -11978,8 +12027,26 @@ private String getPrimaryKeyConstraintName(String catName, String dbName, String
    * @throws MetaException
    */
   @Override
+  @Deprecated
   public SQLAllTableConstraints getAllTableConstraints(String catName, String dbName, String tblName)
       throws MetaException,NoSuchObjectException {
+    AllTableConstraintsRequest request = new AllTableConstraintsRequest(dbName,tblName,catName);
+    return getAllTableConstraints(request);
+  }
+
+  /**
+   * Api to fetch all constraints at once
+   * @param tableConstraintsRequest request object
+   * @return all table constraints
+   * @throws MetaException
+   * @throws NoSuchObjectException
+   */
+  @Override
+  public SQLAllTableConstraints getAllTableConstraints(AllTableConstraintsRequest tableConstraintsRequest)
+      throws MetaException, NoSuchObjectException {
+    String catName = tableConstraintsRequest.getCatName();
```

Comment:
Directly use them from the request object.

Reply:
You can rename tableConstraintsRequest to request, so that the code is more readable.

Reply:
done